index.js

```diff
@@ -519,6 +548,32 @@ class ServerlessAppsyncPlugin {
         config,
         `GraphQlResolver${this.getCfnName(tpl.type)}${this.getCfnName(tpl.field)}`
       );
+      if (tpl.kind === 'PIPELINE') {
+        return Object.assign({}, acc, {
+          [logicalIdResolver]: {
+            Type: 'AWS::AppSync::Resolver',
+            DependsOn: logicalIdGraphQLSchema,
+            Properties: {
+              ApiId: { 'Fn::GetAtt': [logicalIdGraphQLApi, 'ApiId'] },
+              TypeName: tpl.type,
+              FieldName: tpl.field,
+              RequestMappingTemplate: this.processTemplate(requestTemplate, config),
+              ResponseMappingTemplate: this.processTemplate(responseTemplate, config),
+              Kind: 'PIPELINE',
+              PipelineConfig: {
+                Functions: tpl.functions.map(functionAttributeName => {
+                  const logicalIdDataSource = this.getLogicalId(
+                    config,
+                    `GraphQlFunctionConfiguration${this.getCfnName(functionAttributeName)}`
+                  );
+                  return { 'Fn::GetAtt': [logicalIdDataSource, 'FunctionId'] };
+                })
+              }
+            }
+          }
+        });
```

Comment:
@Foosballfan This is not DRY with line 578.
Can we try to reuse what is reusable?

Reply:
Good call, now improved.